ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V108_add_deduplication_info_to_completions.sql

```diff
@@ -0,0 +1,19 @@
+--  Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+--  SPDX-License-Identifier: Apache-2.0
+
+-- Add columns to store command deduplication information in completions.
+
+ALTER TABLE participant_command_completions
+    -- The application ID has to be provided by the application.
+    ADD COLUMN application_id text NOT NULL,
+    -- The submission ID will be provided by the participant or driver if the application didn't provide one.
+    ADD COLUMN submission_id text NOT NULL,
```

Comment:
I guess these two will need to be nullable to accomodate historical data.

Reply:
Yes, absolutely. And I recommend making them nullable on all databases so they're not out of sync.

Reply:
Will this work? How do we populate for old data?

Reply:
Made both nullable in all the schemas: 3e5fa10

Reply:
Actually, application ID was not nullable in the PSQL schema, I was thrown off by a comment there. Reverted to non-nullable in all schemas.

Reply:
(More correctly, it is still nullable because it also contained checkpoints but they were migrated out at some point)